To: vim_dev@googlegroups.com
Subject: Patch 7.3.258
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.258
Problem:    MS-Windows: The edit with existing vim context menu entries can be
	    unwanted.
Solution:   Let a registry entry disable them. (Jerome Vuarand)
Files:	    src/GvimExt/gvimext.cpp


*** ../vim-7.3.257/src/GvimExt/gvimext.cpp	2010-08-15 21:57:29.000000000 +0200
--- src/GvimExt/gvimext.cpp	2011-07-20 16:48:09.000000000 +0200
***************
*** 586,593 ****
  
      // Initialize m_cntOfHWnd to 0
      m_cntOfHWnd = 0;
!     // Retrieve all the vim instances
!     EnumWindows(EnumWindowsProc, (LPARAM)this);
  
      if (cbFiles > 1)
      {
--- 586,608 ----
  
      // Initialize m_cntOfHWnd to 0
      m_cntOfHWnd = 0;
! 
!     HKEY keyhandle;
!     bool showExisting = true;
! 
!     // Check whether "Edit with existing Vim" entries are disabled.
!     if (RegOpenKeyEx(HKEY_LOCAL_MACHINE, "Software\\Vim\\Gvim", 0,
! 				       KEY_READ, &keyhandle) == ERROR_SUCCESS)
!     {
! 	if (RegQueryValueEx(keyhandle, "DisableEditWithExisting", 0, NULL,
! 						 NULL, NULL) == ERROR_SUCCESS)
! 	    showExisting = false;
! 	RegCloseKey(keyhandle);
!     }
! 
!     // Retrieve all the vim instances, unless disabled.
!     if (showExisting)
! 	EnumWindows(EnumWindowsProc, (LPARAM)this);
  
      if (cbFiles > 1)
      {
*** ../vim-7.3.257/src/version.c	2011-07-20 16:36:35.000000000 +0200
--- src/version.c	2011-07-20 16:54:34.000000000 +0200
***************
*** 711,712 ****
--- 711,714 ----
  {   /* Add new patch number below this line */
+ /**/
+     258,
  /**/

-- 
What the word 'politics' means: 'Poli' in Latin meaning 'many' and 'tics'
meaning 'bloodsucking creatures'.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
